### PR TITLE
Restrict backtracking during peep pathfinding

### PR DESCRIPTION
**Pull request by zaxcav**
Dependent on the path layout it is possible for peeps to get stuck in backtracking cycles between two or more path tiles - peeps turn around and backtrack because from those tiles, the reachable tile on the search boundary that is closest to the peep destination is in the direction the peep came from.

In choose_direction when storing/updating the pathfind_history, in addition to removing the chosen direction **also** remove the direction the peep came from from the directions left to try to prevent such backtracking.

Network version is also incremented.

@Broxzier spotted peeps stuck like this in "The OpenRCT2 Group Park 4.58.sv6" and asked if I could take a look.  This PR resolves the issue in that case.

### CODE REVIEW

File: src/openrct2/peep/peep.c [hunk 1]
```diff
@@ -9795,6 +9795,9 @@ int peep_pathfind_choose_direction(sint16 x, sint16 y, uint8 z, rct_peep *peep)
 				/* Peep remembers this junction, so remove the
 				 * chosen_edge from those left to try. */
 				peep->pathfind_history[i].direction &= ~(1 << chosen_edge);
+				/* Also remove the edge the peep came from from
+				 * those left to try. */
```
Comment (Broxzier) [inline src/openrct2/peep/peep.c]: `from from`

File: src/openrct2/peep/peep.c [hunk 2]
```diff
@@ -9813,6 +9816,9 @@ int peep_pathfind_choose_direction(sint16 x, sint16 y, uint8 z, rct_peep *peep)
 		peep->pathfind_history[i].z = z;
 		peep->pathfind_history[i].direction = permitted_edges;
 		peep->pathfind_history[i].direction &= ~(1 << chosen_edge);
+		/* Also remove the edge the peep came from from
+		 * those left to try. */
```
Comment (Broxzier) [inline src/openrct2/peep/peep.c]: `from from`

File: src/openrct2/peep/peep.c (hunk 1 above)
Comment (zaxcav) [inline src/openrct2/peep/peep.c]: @Broxzier The "from from" is intentional.
Would a comma between them be correct grammar?

Comment (janisozaur) [inline src/openrct2/peep/peep.c]: > remove the edge the peep came from **out of** those left to try

> **exclude** the edge the peep came from **out of** those left to try

> remove the **peep's origin(ating) edge** from those left to try

Some ideas about rewording it a little.

Comment (Broxzier) [inline src/openrct2/peep/peep.c]: @zaxcav I see, the comment confused me at first.

Comment (zaxcav) [inline src/openrct2/peep/peep.c]: @Broxzier Yes, that's fair enough. I'm happy to change it to reduce potential confusion.
@janisozaur Thanks for the suggestions.

**Pull request merged**